src/nvim/os/shell.c

```diff
@@ -70,6 +73,25 @@ char **shell_build_argv(const char *cmd, const char *extra_args)
   return rv;
 }
 
+/// Builds the argument vector for any command with with arguments.
+///
+/// Creates: ["command", "that", "has", "spaces"] from a command string
+///
+/// @param cmd Command string
+/// @return A newly allocated argument vector. It must also be freed with
+///         `shell_free_argv` when no longer needed.
+char **build_argv(char *cmd)
+  FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC
+{
+  size_t argc = tokenize((char_u *)cmd, NULL);
+  char **rv = xmalloc((argc + 4) * sizeof(*rv));
+
+  tokenize((char_u *)cmd, rv);
+  assert(rv[0]);
```

Comment:
An empty string is not a `NULL` pointer, but has its first char set to `NUL`. For such a command string
`tokenize()` only returns `0` without writing into `rv`. Could happen with `:set sh=`.